### PR TITLE
feat: jdtls per project directory under ~/workspace

### PR DESCRIPTION
**Pull request by ChristianChiarulli**

### CODE REVIEW

File: lua/nvim-lsp-installer/servers/jdtls/init.lua [hunk 1]
```diff
@@ -10,6 +10,7 @@ return function(name, root_dir)
         local executable = vim.env.JAVA_HOME and path.concat { vim.env.JAVA_HOME, "bin", "java" } or "java"
         local jar = vim.fn.expand(path.concat { root_dir, "plugins", "org.eclipse.equinox.launcher_*.jar" })
         local lombok = vim.fn.expand(path.concat { root_dir, "lombok.jar" })
+        local workspace_dir = vim.fn.fnamemodify(vim.fn.getcwd(), ":p:h:t")
```
Comment (williamboman) [inline lua/nvim-lsp-installer/servers/jdtls/init.lua:13]: Wouldn't the following do it?

```suggestion
        local workspace_dir = vim.fn.fnamemodify(vim.fn.getcwd(), ":t")
```

File: lua/nvim-lsp-installer/servers/jdtls/init.lua [hunk 2]
```diff
@@ -38,7 +39,7 @@ return function(name, root_dir)
                 ),
             },
             "-data",
-            workspace_name,
+            workspace_name .. workspace_dir,
```
Comment (williamboman) [inline lua/nvim-lsp-installer/servers/jdtls/init.lua:42]:
```suggestion
            patch.concat { workspace_name, workspace_dir },
```

File: lua/nvim-lsp-installer/servers/jdtls/init.lua [hunk 3]
```diff
@@ -57,7 +58,7 @@ return function(name, root_dir)
             std.download_file("https://projectlombok.org/downloads/lombok.jar", "lombok.jar"),
         },
         default_options = {
-            cmd = get_cmd(vim.env.WORKSPACE and vim.env.WORKSPACE or path.concat { vim.env.HOME, "workspace" }),
+            cmd = get_cmd(vim.env.WORKSPACE and vim.env.WORKSPACE or path.concat { vim.env.HOME, "workspace/" }),
```
Comment (williamboman) [inline lua/nvim-lsp-installer/servers/jdtls/init.lua:61]:
```suggestion
            cmd = get_cmd(vim.env.WORKSPACE and vim.env.WORKSPACE or path.concat { vim.env.HOME, "workspace" }),
```

**Review by williamboman — COMMENTED** [review]
Thanks! Again this is mostly mirrored off of lspconfig, so I think this ought to be patched in lspconfig too

File: lua/nvim-lsp-installer/servers/jdtls/init.lua (hunk 2 above)
Comment (ChristianChiarulli) [reply lua/nvim-lsp-installer/servers/jdtls/init.lua:42]: should this be patch.concat or path.concat?

File: lua/nvim-lsp-installer/servers/jdtls/init.lua (hunk 1 above)
Comment (ChristianChiarulli) [reply lua/nvim-lsp-installer/servers/jdtls/init.lua:13]: Not sure, nvim-jdtls decided to do it this way and I trusted them: https://github.com/mfussenegger/nvim-jdtls

File: lua/nvim-lsp-installer/servers/jdtls/init.lua (hunk 3 above)
Comment (ChristianChiarulli) [reply lua/nvim-lsp-installer/servers/jdtls/init.lua:61]: once we sort out my other question we can commit this

File: lua/nvim-lsp-installer/servers/jdtls/init.lua (hunk 2 above)
Comment (williamboman) [reply lua/nvim-lsp-installer/servers/jdtls/init.lua:42]: `path.concat`!

**Review by williamboman — APPROVED** [review]

**Comment by williamboman** [comment]
Thanks!!

**Pull request merged**